.github/workflows/internal-images.yml

```diff
@@ -43,10 +43,23 @@ jobs:
           #   - needs repo:status, public_repo, write:packages, delete:packages
           password: ${{ secrets.GHCR_TOKEN }}
 
-      - name: Build and push
-        run: |  # This will only push a single architecture, which is fine as we currently only support amd64
+      # We need QEMU and Buildx for multi-platform (amd64+arm64) image push. See RATIONALE.md
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Setup Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Write Dockerfile
+        run: |
           cat > Dockerfile <<'EOF'
           ${{ matrix.dockerfile }}
           EOF
-          docker build -t ghcr.io/tetratelabs/func-e-internal:${{ matrix.target_tag }} .
-          docker push ghcr.io/tetratelabs/func-e-internal:${{ matrix.target_tag }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
```

Comment:
ps this is more opaque than I'd like as it is really hard to test what this might do locally (presumably buildx commands). Meanwhile we can use the action until it proves itself a nuisance (which might not happen ever if we are lucky!)